### PR TITLE
build: stop creating files as root in verify-archive.sh

### PR DESCRIPTION
**Pull request by benesch**
If we build in a folder shared with the host, Go creates files in that
are owned by root. Use a temporary directory instead.

### CODE REVIEW

File: build/verify-archive.sh [hunk 1]
```diff
@@ -10,8 +10,9 @@ set -euo pipefail
 apt-get update
 apt-get install -y cmake xz-utils
 
-tar xzf cockroach.src.tgz
-(cd cockroach-* && make install)
+workdir="$(mktemp -d)"
```
Comment (tamird) [inline build/verify-archive.sh]: my shellcheck-fu is a little rust, but I believe you don't need the quotes here.

probably best to consult shellcheck, though.

Comment (tamird) [inline build/verify-archive.sh]: Don't you need to remove the directory when you're done?

**Review by tamird — APPROVED** [review]
Docker-is-root is the gift that keeps on giving.

File: build/verify-archive.sh (hunk 1 above)
Comment (jordanlewis) [inline build/verify-archive.sh]: This whole file is run within Docker, so theoretically the files will be dropped on container exit, right?

On the other hand, an explicit `rm -rf $workdir` in an exit trap might be nice for readability.

**Review by jordanlewis — APPROVED** [review]

File: build/verify-archive.sh (hunk 1 above)
Comment (tamird) [inline build/verify-archive.sh]: Good point. That reminds me, the current `docker run` invocation in `build/teamcity-verify-archive.sh` doesn't contain `--rm`, which means we're keeping these containers indefinitely. That will definitely have a bad impact on disk usage, so we should fix it before we enable the build step in TC.

Comment (benesch) [inline build/verify-archive.sh]: Done.

Comment (benesch) [inline build/verify-archive.sh]: They are unnecessary, but I do it out of habit for consistency. Don't feel strongly, so I dropped the quotes.

**Pull request merged**